module/VuFind/src/VuFind/RecordDriver/AbstractBase.php

```diff
@@ -369,65 +388,70 @@ protected function expandLegacyAbbreviation(string $abbr) {
      */
     public function getCitationFormats()
     {
-        $formatSetting = $this->mainConfig->Record->citation_formats ?? true;
+        $formatSetting = $this->mainConfig->Record->get('citation_formats', true);
 
-        // Citations disabled:
+        // Citations disabled
         if ($formatSetting === false || $formatSetting === 'false') {
             return [];
+        // Legacy: use all supported options.
+        } else if ($formatSetting === true || $formatSetting === 'true') {
+            // Defaults
+            $formatSetting = [
+                'APA Citation:apa',
+                'Chicago Style Citation:chicago-annotated-bibliography',
+                'MLA Citation:modern-language-association',
+            ];
+        // Legacy: convert to array
+        } else if (is_string($formatSetting)) {
+            $formatSetting = explode(',', $formatSetting);
+        } else if (!is_array($formatSetting)) {
+            $formatSetting = $formatSetting->toArray();
         }
 
-        // Default behavior: use all supported options.
-        if ($formatSetting === true || $formatSetting === 'true') {
-            $formatSetting = $this->tryMethod(
-                'getDefaultCitationFormats',
-                [], // ← params, ↓ fallback to legacy
-                $this->getSupportedCitationFormats()
-            );
+        // Trim and convert legacy to 9.x format
+        foreach ($formatSetting as $i => $format) {
+            $formatSetting[$i] = trim($this->expandLegacyAbbreviation($format));
```

Comment:
It might make sense to create an anonymous function like:

<pre>
$formatCleanup = function ($format) {
    return trim($this->expandLegacyAbbreviation($format));
}
</pre>

...then you can replace the foreach loops here and below with `array_map` calls, which might make the code a little more concise (as well as ensuring that format reformatting is always done consistently).

Reply:
I could also make the trim part of the expandLegacyAbbreviation function.

Reply:
True, that might be simpler, and then you could array_map with the `[$this, 'expandLegacyAbbreviation']` callback.